v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/migrations/avro/AvroToValueMapperTest.java

```diff
@@ -168,6 +168,44 @@ public void testAvroFieldToDouble_UnsupportedType() {
     AvroToValueMapper.avroFieldToDouble(inputValue, SchemaBuilder.builder().booleanType());
   }
 
+  @Test
+  public void testAvroFieldToFloat_ValidFloatInput() {
+    Float inputValue = 5.75f;
+    Float result =
+        AvroToValueMapper.avroFieldToFloat(inputValue, SchemaBuilder.builder().doubleType());
+    assertEquals(inputValue, result);
+
+    /* Test a valid double to float mapping */
+    result = AvroToValueMapper.avroFieldToFloat((Double) 3.14, SchemaBuilder.builder().floatType());
+    assertEquals("Test float input", (Float) 3.14f, result);
+
+    result = AvroToValueMapper.avroFieldToFloat("456.346", SchemaBuilder.builder().doubleType());
+    assertEquals("Test string input", (Float) 456.346f, result);
+
+    result = AvroToValueMapper.avroFieldToFloat(Long.MAX_VALUE, SchemaBuilder.builder().longType());
```

Comment:
Do we want to add some edge cases for float testing the limits of the range? If we have some for double, we should add that too, else we can maybe create a ticket to track a todo.

Reply:
We dont have for double. But I can try for float.